codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Connection.java

```diff
@@ -83,9 +85,14 @@
         void onWeightChanged(Http2Stream stream, short oldWeight);
 
         /**
-         * Called when a GO_AWAY frame has either been sent or received for the connection.
+         * Called when a GO_AWAY frame was sent for the connection.
+         */
+        void goAwaySent(long errorCode, ByteBuf debugData);
+
+        /**
+         * Called when a GO_AWAY frame was received for the connection.
          */
-        void goingAway();
+        void goAwayReceived(long errorCode, ByteBuf debugData);
```

Comment:
@Scottmitch It's clear that `Http2FrameListener` needs to have `onGoAwayRead`. This interface is rather raw and perhaps it shouldn't be exposed beyond its use internally within the decoder.  Maybe all the events that we want exposed should be moved (one way or another) into `Connection.Listener`.

Having these two types of listeners has been a pain point for a while and it's probably time that we sorted it out.  WDYT?